src/main/java/org/jenkinsci/plugins/ssegateway/sse/EventDispatcher.java

```diff
@@ -232,6 +263,105 @@ private void publishStateEvent(SSEChannel.Event event, Message additional) {
             LOGGER.log(Level.WARNING, "Failed to publish SSE Dispatcher state event.", e);
         }
     }
+
+    private void dispatchReload() {
+        retryQueue.clear();
+        try {
+            dispatchEvent("reload", null);
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Unable to send reload event to client.", e);
+        }
+    }
+    
+    private void addToRetryQueue(@Nonnull Message message) {
+        if (!retryQueue.add(new Retry(message))) {
+            // Unable to add to the queue. Lets just tell the client
+            // that it needs to reload the page.
+            dispatchReload();
+        }
+    }
+
+    synchronized void processRetries() {
+        Retry retry = retryQueue.peek();
+        
+        try {
+            while (retry != null) {
+                try {
+                    String eventJSON = EventHistoryStore.getChannelEvent(retry.channelName, retry.eventUUID);
+
+                    if (eventJSON == null) {
+                        // The event is not in the store. This can be simply because the event has
+                        // not yet arrived at the store and been stored. It might need another
+                        // moment or two to get there.
+                        if (!retry.needsMoreTimeToLandInStore()) {
+                            // Something's gone wrong. The event should be in the store by now. 
+                            // Lets tell the client that it needs to do a full page reload. Not much
+                            // else can be done at this stage.
+                            dispatchReload(); // This clears the queue too.
+                            return;
+                        } else {
+                            // The event should be in the store (not expired, so would not 
+                            // have been deleted). Only explanation is that it has not yet
+                            // landed there. Let's pause the retry process
+                            // for a moment and retry again in the hope that the event
+                            // eventually lands there. Exiting here will schedule a new run
+                            // of this retry process (see finally block below).
+                            return;
+                        }
+                    }
+                    
+                    if (Util.isTestEnv()) {
+                        JSONObject eventJSONObj = JSONObject.fromObject(eventJSON);
+                        eventJSONObj.put(SSEChannel.EventProps.sse_dispatch_retry.name(), "true");
+                        eventJSON = eventJSONObj.toString();
+                    }
+
+                    if (!dispatchEvent(retry.channelName, eventJSON)) {
+                        LOGGER.log(Level.FINE, String.format("Error dispatching retry event to SSE channel. Write failed. Dispatcher %s.", this));
+                        return;
```

Comment:
Could this get into an infinite loop since not removed here? Or is that only for the max age of the queue?